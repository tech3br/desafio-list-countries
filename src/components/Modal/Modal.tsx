import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogTitle from "@material-ui/core/DialogTitle";
import Slide from "@material-ui/core/Slide";
import { TransitionProps } from "@material-ui/core/transitions";
import React from "react";

interface Props {
  open: boolean;
  textEdit: string;
  textClose: string;
  titleModal: string;
  contentModal: string;
  handleClose(): void;
  handleEdit(): void;
}

//transicao no modal
const Transition = React.forwardRef(function Transition(
  props: TransitionProps & { children?: React.ReactElement<any, any> },
  ref: React.Ref<unknown>
) {
  return <Slide direction="down" ref={ref} {...props} />;
});

const ModalComponent = (props: Props) => {
  return (
    <div>
      <Dialog
        open={props.open}
        TransitionComponent={Transition}
        keepMounted
        aria-labelledby="alert-dialog-slide-title"
        aria-describedby="alert-dialog-slide-description"
      >
        <DialogTitle id="alert-dialog-slide-title">
          {props.titleModal}
        </DialogTitle>
        <DialogContent>
          <DialogContentText id="alert-dialog-slide-description">
            {props.contentModal}
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button color="primary" onClick={props.handleEdit}>
            {props.textEdit}
          </Button>
          <Button color="primary" onClick={props.handleClose}>
            {props.textClose}
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};

export default ModalComponent;
