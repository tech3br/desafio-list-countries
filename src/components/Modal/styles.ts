import { makeStyles } from "@material-ui/core";

export const styles = makeStyles({
  container: {
    backgroundColor: "#000",
  },
});
