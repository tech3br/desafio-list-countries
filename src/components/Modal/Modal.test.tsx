import { render } from "@testing-library/react";
import React from "react";
import Modal from "./Modal";

describe("Testando o componente Modal", () => {
  it("should render correctly", () => {
    const wrapper = render(
      <Modal
        open
        textEdit={"editar"}
        textClose={"fechar"}
        handleClose={() => {}}
        handleEdit={() => {}}
        titleModal={"teste"}
        contentModal={"teste"}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
