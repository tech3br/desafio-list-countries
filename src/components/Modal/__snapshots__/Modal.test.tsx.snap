// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testando o componente Modal should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 0px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    >
      <div />
    </div>
    <div
      class="MuiDialog-root"
      role="presentation"
      style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper"
        role="none presentation"
        style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-describedby="alert-dialog-slide-description"
          aria-labelledby="alert-dialog-slide-title"
          class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
          role="dialog"
        >
          <div
            class="MuiDialogTitle-root"
            id="alert-dialog-slide-title"
          >
            <h2
              class="MuiTypography-root MuiTypography-h6"
            >
              teste
            </h2>
          </div>
          <div
            class="MuiDialogContent-root"
          >
            <p
              class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
              id="alert-dialog-slide-description"
            >
              teste
            </p>
          </div>
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                editar
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                fechar
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
